server/src/modules/mixin/rebalance/rebalance.service.ts

```diff
@@ -221,12 +221,12 @@ export class RebalanceService {
     const minBalance = new BigNumber(minimum_balance);
 
     // Determine if rebalance is necessary
-    if (balanceBN.isLessThanOrEqualTo(minBalance)) {
+    if (!balanceBN.isLessThanOrEqualTo(minBalance)) {
       return;
     }
-
+    console.warn(balanceBN.minus(minBalance).toString());
     // Calculate the amount needed to rebalance
-    const amountToRebalance = balanceBN.minus(minBalance);
+    const amountToRebalance = minBalance.minus(balanceBN);
```

Comment:
I'm sorry the previous logic is wrong here. The modification is right under the previous context, but the previous logic is wrong. If we apply the `amountToRebalance` here, the rebalance amount would be minor and won't cover the withdrawal cost. it would be a waste.

What we expect is to get the `amountToRebalance` based on the `mixinBalance` and `exchangeBalance`. The parameter `balance` in `rebalanceFromMixinToExchange()` is the `exchangeBalance`. But the `mixinBalance` is not here. We will need to modify it, and calculate the `amountToRebalance` based on both value.

Reply:
@mj-blockydevs Here we need some modification. Sorry for the bugs I wrote.